components/webext-storage/sql/create_sync_temp_tables.sql

```diff
@@ -28,7 +28,27 @@ CREATE TEMP TABLE IF NOT EXISTS storage_sync_outgoing_staging (
     guid TEXT NOT NULL PRIMARY KEY DEFAULT(generate_guid()),
     ext_id TEXT NOT NULL UNIQUE,
     data TEXT,
-    sync_change_counter INTEGER NOT NULL
+    sync_change_counter INTEGER NOT NULL,
+    was_uploaded BOOLEAN NOT NULL DEFAULT 0
 );
 
+CREATE TEMP TRIGGER IF NOT EXISTS record_uploaded
```

Comment:
ah, I was wondering how you envisaged the trigger working when I updated my patch yesterday - this makes great sense.